src/shared/components/QueryBuilder/ui/inputs/SingleEditionInput.tsx

```diff
@@ -0,0 +1,104 @@
+import { css, cx } from '@emotion/css';
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import { useStyles2 } from '@grafana/ui';
+import React, { useEffect, useRef } from 'react';
+
+type SingleEditionInputProps = {
+  placeholder: string;
+  onChange: (suggestion: SelectableValue<string>) => void;
+  ref?: React.Ref<HTMLInputElement>;
+  defaultValue?: string;
+  onFocus?: () => void;
+};
+
+export function SingleEditionInput({ placeholder, defaultValue, onFocus, onChange }: SingleEditionInputProps) {
+  const styles = useStyles2(getStyles);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const onKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    if (e.code === 'Enter') {
+      const value = (e.target as HTMLInputElement).value.trim();
+      onChange({ value, label: value });
+    }
+  };
+
+  const onBlur = (e: React.FocusEvent<HTMLInputElement>) => {
+    const value = (e.target as HTMLInputElement).value.trim();
+    onChange({ value, label: value });
+  };
+
+  useEffect(() => {
+    if (inputRef.current) {
+      inputRef.current.focus();
+    }
+  });
+
+  // we have to implement a customized version of the input because the <Input /> from the Saga Design system does not
```

Comment:
Having a separate component should be fine, though it seems the Input forwards the ref in https://github.com/grafana/grafana/blob/47f87171496f675bc44ddf839effaed1846bfa4e/packages/grafana-ui/src/components/Input/Input.tsx#L34 and the legacy input allows `inputRef` property https://github.com/grafana/grafana/blob/47f87171496f675bc44ddf839effaed1846bfa4e/packages/grafana-ui/src/components/Forms/Legacy/Input/Input.tsx#L85

Reply:
Funny how I missed it. I'll have a look in the next PR. Thanks